compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -670,10 +679,15 @@ static void PrintService(const ServiceDescriptor* service,
   (*vars)["service_name"] = service->name();
   (*vars)["file_name"] = service->file()->name();
   (*vars)["service_class_name"] = ServiceClassName(service);
+  #ifdef GRPC_VERSION
+    (*vars)["grpc_version"] = "(version "+string(XSTR(GRPC_VERSION))+")";
+  #else
+    (*vars)["grpc_version"] = "";
+  #endif
   p->Print(
       *vars,
       "@$Generated$(\n"
-      "    value = \"by gRPC proto compiler\",\n"
+      "    value = \"by gRPC proto compiler $grpc_version$\",\n"
```

Comment:
nit: This leaves a trailing space after compiler if grpc_version isn't used. That space would ideally be included in the grpc_version variable itself and have no space here. Since this is only seen by humans it isn't a big deal...